Automattic/wp-calypso
# Sites: only display hosting flow forking empty state while in the hosting flow

## Proposed Changes

We explored new developer account ideas, including always sending the user to the `/setup/new-hosted-site` flow in order to always create advanced sites for them, or at least to lower the number of required steps before having a site ready to be tweaked.

When coming from `/hosting`, users go through `/setup/new-hosted-site` as their onboarding flow. The flow is minimal (on purpose): it asks for a site title and a data center, and automatically transfers Business and Commerce sites to Atomic after checkout so that the user can start tweaking the site right away.

This is fine for onboarding, but there's a problem with this approach for subsequent purchases. If you tick the developer checkbox on `/start/hosting`, the "Add new site" button will always send you to `/setup/new-hosted-site`. This is because we assumed that developers would always want to create advanced sites, with custom themes and plugins. That means that, as a developer, you'll never be able to create a blog or a newsletter, for example.

We should move away from fragmenting the experience. We have plans to, in the future, introduce a Build site intent so that we can let the user create sites, be it a newsletter or a customized hosted site, through the intent-picker.

To top it, on a semi-A/B test with developers and non-developers and data from last week, it turns out that the `/setup/new-hosted-site` flow actually converts worse than the `/start` flow:

- 3.2% for "Add new site" -> `/setup/new-hosted-site` (tracks/funnel/view/?44187/2023-07-12/to/2023-07-18);
- 4.9% for "Add new site" -> `/start` (tracks/funnel/view.php?44188/2023-07-12/to/2023-07-18).

These changes restore the initial state from /sites and removes the branching from the hosted flow sites, simplifying the implementation.

| /sites when the user is in the hosting flow | /sites when the user is not in the hosting flow |
| ----------------------------------------- | --------------------------------------------- |
| ![image](https://github.com/Automattic/wp-calypso/assets/26530524/173d0ccb-0598-4279-a081-1062d1788480) | ![image](https://github.com/Automattic/wp-calypso/assets/26530524/48de146f-9385-4f22-93b4-0a0a4c463e4b) |

## Testing Instructions

1. Open `/sites` and you should verify that you see the empty sites placeholder in the sites page, and not the hosting flow forking state;
2. Open `/sites?hosting-flow=true` and you should see the hosting flow forking component instead of the sites page;
3. Click on "Create a site" and verify that you're sent to `/setup/new-hosted-site` (without the `?hosting-flow=true` param) and you see the options step with a data center selector.
